src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
 
-def stop_daos_agent_services(test_file, args):
-    """Stop any daos_agent.service running on the hosts running servers.
+        """
+        source_files = os.path.join(source, pattern)
+        logger.debug("-" * 80)
+        logger.debug(
+            "Checking for any %s files exceeding %s on %s", source_files, threshold, hosts)
+        other = ["-size", f"+{threshold}", "-printf", "'%p %k KB'"]
+        result = run_remote(logger, hosts, find_command(source, pattern, depth, other))
+        if not result.passed:
+            message = f"Error checking for {source_files} files exceeding the {threshold} threshold"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            return 32
+
+        # The command output will include the source path if the threshold has been exceeded
+        for data in result.output:
+            if source in "\n".join(data.stdout):
+                message = f"One or more {source_files} files exceeded the {threshold} threshold"
+                self._fail_test(self.result.tests[-1], "Process", message)
+                return 32
+
+        logger.debug("No %s file sizes found exceeding the %s threshold", source_files, threshold)
+        return 0
+
+    def _cart_log_test(self, hosts, source, pattern, depth):
+        """Run cart_logtest on the log files.
 
-    Args:
-        test_file (dict): a dictionary of the test script/yaml file
-        args (argparse.Namespace): command line arguments for this program
+        Args:
+            hosts (NodSet): hosts on which the files are located
+            source (str): where the files are currently located
+            pattern (str): pattern used to limit which files are processed
+            depth (int): max depth for find command
 
-    Returns:
-        int: status code: 0 = success, 512 = failure
+        Returns:
+            int: status code: 0 = success, 16 = failure
 
-    """
-    service = "daos_agent.service"
-    hosts = get_hosts_from_yaml(test_file["yaml"], args, YAML_KEYS["test_clients"])
-    hosts.add(NodeSet(socket.gethostname().split('.', 1)[0]))
-    log("-" * 80)
-    log("Verifying {} on {} after running '{}'".format(service, hosts, test_file["py"]))
-    return stop_service(hosts, service)
+        """
+        source_files = os.path.join(source, pattern)
+        cart_logtest = os.path.abspath(os.path.join("cart", "cart_logtest.py"))
+        logger.debug("-" * 80)
+        logger.debug("Running %s on %s files on %s", cart_logtest, source_files, hosts)
+        other = ["-print0", "|", "xargs", "-0", "-r0", "-n1", "-I", "%", "sh", "-c",
+                 f"'{cart_logtest} % > %.cart_logtest 2>&1'"]
+        result = run_remote(logger, hosts, find_command(source, pattern, depth, other), timeout=900)
+        if not result.passed:
+            message = f"Error running {cart_logtest} on the {source_files} files"
+            self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call